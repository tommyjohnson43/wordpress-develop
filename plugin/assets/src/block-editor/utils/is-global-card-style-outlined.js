/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { getConfig } from './index';

/**
 * Whether global card style outlined.
 *
 * @return {boolean} Is outlined.
 */
const isGlobalCardStyleOutlined = () =>
	// eslint-disable-next-line camelcase
	getConfig( 'defaults' )?.globalStyle?.card_style === 'outlined';

const getGlobalCardStyle = () =>
	// eslint-disable-next-line camelcase
	getConfig( 'defaults' )?.globalStyle?.card_style;

export { isGlobalCardStyleOutlined, getGlobalCardStyle };
