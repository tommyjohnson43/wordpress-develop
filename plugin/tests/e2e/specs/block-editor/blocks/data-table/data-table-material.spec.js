/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { createNewPost, getEditedPostContent } from '@wordpress/e2e-test-utils';
import { insertBlockByKeyword } from '../../../../utils';

describe( 'blocks: material/data-table', () => {
	beforeEach( async () => {
		await createNewPost( {} );
		await insertBlockByKeyword( 'Material Data Table' );
	} );

	it( 'should be inserted', async () => {
		// Check if block was inserted
		expect(
			await page.$( '[data-type="material/data-table"]' )
		).not.toBeNull();
	} );

	// eslint-disable-next-line jest/no-disabled-tests
	it.skip( 'should update table cell content', async () => {
		const createButton = await page.$(
			'[data-type="material/data-table"] form [type="submit"]'
		);
		createButton.click();
		await page.waitForSelector( 'td' );
		await page.click( 'td' );
		await page.keyboard.type( 'Column 1' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
