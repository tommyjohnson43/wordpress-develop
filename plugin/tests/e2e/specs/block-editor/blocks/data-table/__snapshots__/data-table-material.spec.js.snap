// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blocks: material/data-table should update table cell content 1`] = `
"<!-- wp:material/data-table -->
<div class=\\"wp-block-material-data-table wp-block-table\\"><div class=\\"mdc-data-table\\"><table class=\\"mdc-data-table__table\\"><tbody class=\\"mdc-data-table__content\\"><tr class=\\"mdc-data-table__row\\"><td class=\\"mdc-data-table__cell\\">Column 1</td><td class=\\"mdc-data-table__cell\\"></td></tr><tr class=\\"mdc-data-table__row\\"><td class=\\"mdc-data-table__cell\\"></td><td class=\\"mdc-data-table__cell\\"></td></tr></tbody></table></div></div>
<!-- /wp:material/data-table -->"
`;
